packages/jaeger-ui/src/components/TracePage/CriticalPath/utils/sanitizeOverFlowingChildren.tsx

```diff
@@ -0,0 +1,100 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Span } from '../../../../types/trace';
+
+// This function truncates/drops the overflowing child spans
+const sanitizeOverFlowingChildren = (spans: Span[]): Span[] => {
+  let sanitizedSpanData: Span[] = [];
+
+  spans.forEach(span => {
+    if (!span.references.length) {
+      sanitizedSpanData.push(span);
+    } else {
+      // parentSpan will be undefined when its parentSpan is dropped previously
+      const parentSpan = sanitizedSpanData.find(s => s.spanID === span.references[0].spanID)!;
+      if (parentSpan) {
+        const childEndTime = span.startTime + span.duration;
+        const parentEndTime = parentSpan.startTime + parentSpan.duration;
+        switch (true) {
+          case span.startTime >= parentSpan.startTime && childEndTime <= parentEndTime:
+            // case 1: everything looks good
+            // |----parent----|
+            //   |----child--|
+            sanitizedSpanData.push(span);
+            break;
+
+          case span.startTime < parentSpan.startTime &&
+            childEndTime <= parentEndTime &&
+            childEndTime > parentSpan.startTime:
+            // case 2: child start before parent, truncate is needed
+            //      |----parent----|
+            //   |----child--|
+            sanitizedSpanData.push({
+              ...span,
+              startTime: parentSpan.startTime,
+              duration: childEndTime - parentSpan.startTime,
+            });
+            break;
+
+          case span.startTime >= parentSpan.startTime &&
+            childEndTime > parentEndTime &&
+            span.startTime < parentEndTime:
+            // case 3: child end after parent, truncate is needed
+            //      |----parent----|
+            //              |----child--|
+            sanitizedSpanData.push({
+              ...span,
+              duration: parentEndTime - span.startTime,
+            });
+            break;
+
+          case span.startTime >= parentEndTime || childEndTime <= parentSpan.startTime:
+            // case 4: child outside of parent range => drop the child span
+            //      |----parent----|
+            //                        |----child--|
+            // or
+            //                      |----parent----|
+            //       |----child--|
+
+            // Remove the childSpanId from its parent span
+            sanitizedSpanData.forEach(each => {
+              if (each.spanID === span.references[0].spanID) {
+                const index = each.childSpanIds.findIndex(id => id === span.spanID);
+                each.childSpanIds.splice(index, 1);
+              }
+            });
+            break;
+
+          default:
+            // Never reaches to default
+            // Something unexpected happened
+            break;
```

Comment:
```suggestion
          default:
            // Never reaches to default
            // Something unexpected happened
            throw RangeError(`Error while computing Critical Path Algorithm.`)
            break;
```

Reply:
You can add more debug values, such as the parent/child span's start/end time.

Reply:
nit: break after throw is redundant